platform/linux-generic/arch/aarch64/odp_crypto_armv8.c

```diff
@@ -0,0 +1,648 @@
+/* Copyright (c) 2014-2018, Linaro Limited
+ * Copyright (c) 2021, ARM Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#include <odp_posix_extensions.h>
+#include <odp/api/crypto.h>
+#include <odp_init_internal.h>
+#include <odp/api/spinlock.h>
+#include <odp/api/sync.h>
+#include <odp/api/debug.h>
+#include <odp/api/align.h>
+#include <odp/api/shared_memory.h>
+#include <odp_debug_internal.h>
+#include <odp/api/hints.h>
+#include <odp/api/random.h>
+#include <odp/api/plat/packet_inlines.h>
+#include <odp/api/plat/thread_inlines.h>
+#include <odp_packet_internal.h>
+#include <odp/api/plat/queue_inlines.h>
+
+#include "AArch64cryptolib.h"
+
+/* Inlined API functions */
+#include <odp/api/plat/event_inlines.h>
+
+#define MAX_SESSIONS 4000
+/* Length in bytes */
+#define ARM_CRYPTO_MAX_CIPHER_KEY_LENGTH      64
+#define ARM_CRYPTO_MAX_AUTH_KEY_LENGTH        64
+#define ARM_CRYPTO_MAX_IV_LENGTH              16
```

Comment:
Max IV length of 16 is larger the actual maximum in the next patch which is 12. But the ARM crypto library wants to read the IV in a 16 byte chunk, so maybe that is why this is 16. It would be good to document such dependency and check the IV length the same way as was done with the max data length.

I do not see why the max cipher key length needs to be 64 since the max key length in the next patch is only 32, but maybe the max cipher and auth key lenght values are preparation for additional algorithms that get introduced later?

Reply:
I will add a similar comment and assert for `ARM_CRYPTO_MAX_IV_LENGTH` as we have done for `ARM_CRYPTO_MAX_DATA_LENGTH`.
As for the ARM_CRYPTO_MAX_CIPHER_KEY_LENGTH, I will change its value to 32. The other algorithms that we want to add, namely, AES-CBC128 + HMAC-SHA1 and AES-CBC128  + HMAC-SHA256 also have only a 16-byte cipher key so it will be taken care of as well.